### PR TITLE
feat: add resource transfer logic to support different ACL strategy

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/RFE-530
https://oat-sa.atlassian.net/browse/ADF-1424
https://oat-sa.atlassian.net/browse/ADF-1425
https://oat-sa.atlassian.net/browse/ADF-1426
https://oat-sa.atlassian.net/browse/ADF-1427

# Goal

Add dynamic ACL strategy to `Move to` and `Copy to` features

# How to test

- Login to backoffice
- Play with the new ACL options for Items/Assets/Tests

# Related PRs

- https://github.com/oat-sa/tao-core/pull/3776
- https://github.com/oat-sa/extension-tao-mediamanager/pull/462
- https://github.com/oat-sa/extension-tao-test/pull/438
- https://github.com/oat-sa/extension-tao-dac-simple/pull/216
- https://github.com/oat-sa/generis/pull/1036

### CODE REVIEW

File: models/classes/Copier/ItemClassCopier.php [hunk 1]
```diff
@@ -26,22 +26,25 @@
...
-class ItemClassCopier implements ClassCopierInterface
+class ItemClassCopier implements ClassCopierInterface, ResourceTransferInterface
 {
-    /** @var ClassCopierInterface */
+    /** @var ClassCopierInterface|ResourceTransferInterface */
     private $taoClassCopier;
+    private Ontology $ontology;
 
-    public function __construct(ClassCopierInterface $taoClassCopier)
+    public function __construct(ClassCopierInterface $taoClassCopier, Ontology $ontology)
```
Comment (gabrielfs7) [inline models/classes/Copier/ItemClassCopier.php:42]: - [ ] Use only ResourceTransferInterface instead

File: models/classes/Copier/ItemClassCopier.php [hunk 2]
```diff
@@ -51,6 +54,13 @@ public function copy(
         return $this->taoClassCopier->copy($class, $destinationClass);
     }
 
+    public function transfer(ResourceTransferCommand $command): ResourceTransferResult
```
Comment (gabrielfs7) [inline models/classes/Copier/ItemClassCopier.php:57]: - [ ] Cover with tests

Comment (gabrielfs7) [reply models/classes/Copier/ItemClassCopier.php:48]: done

File: models/classes/Copier/ItemClassCopier.php (hunk 1 above)
Comment (gabrielfs7) [reply models/classes/Copier/ItemClassCopier.php]: done

File: composer.json [hunk 3]
```diff
@@ -67,7 +67,7 @@
     "require": {
         "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
         "oat-sa/generis": ">=15.22",
-        "oat-sa/tao-core": ">=50.26.0",
+        "oat-sa/tao-core": "dev-feat/RFE-530/integration-branch as 52.0",
```
Comment (gabrielfs7) [inline composer.json:70]: - [ ] Use proper version after release

File: composer.json [hunk 4]
```diff
@@ -66,8 +60,8 @@
     },
     "require": {
         "oat-sa/oatbox-extension-installer": "~1.1||dev-master",
-        "oat-sa/generis": ">=15.22",
-        "oat-sa/tao-core": ">=50.26.0",
+        "oat-sa/generis": ">=15.23",
+        "oat-sa/tao-core": "dev-feat/RFE-530/integration-branch as v52.0.0",
```
Comment (hectoras) [inline composer.json:64]: - [ ] Don't forget to update this :)

**Review by shpran — APPROVED** [review]
- [x] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic
- [x] Changelog is updated according to changes (if applicable)
- [x] Documentation is updated according to changes (if applicable)

**Pull request merged**